src/beacon/beacon_entries.rs

```diff
@@ -33,6 +33,14 @@ pub struct BeaconEntry {
 
 impl BeaconEntry {
     pub fn new(round: u64, signature: Vec<u8>) -> Self {
+        // Drop any excess capacity to make heap usage deterministic across allocators.
+        // Avoid shrink_to_fit: it's a non-binding hint.
```

Comment:
What does a "non-binding hint" means?

Reply:
It means not all allocators respect `shrink_to_fit` to rellocate the Vec and make capacity == len